### DISCUSSION TITLE
Qwik support?

### DISCUSSION THREAD

**Discussion by redbar0n** [tamagui/tamagui#497] [Q&A]
Would it be feasible to use Tamagui with Qwik?

As you know, Qwik enables use of the React ecosystem through ‘qwikifying’ React components.[1]

Qwik has some styling limitations / peculiarities, to enable lazy loading:

https://qwik.builder.io/docs/components/styles/#usestyles

From that page, you see any potential blockers @natew ?

Qwik is much more compelling than using React Server Components. Since Qwik is optimized by default through progressively downloading JS component code. Whereas RSC only execute on server, transfer the template repeatedly, and developers have to decide for every component what type it should be. There’s also the issue of SSR hydration, which NextJS has to perform, but Qwik can merely resume the work the server already did (effectively avoiding the double work), so Qwik doesn’t put as much load on the client: good for emerging markets, 3G and low end Android telephones, and the web in general.

Some like @angel-rs have tried to use Tamagui with Qwik, but had to resort to Nativewind instead, since Qwik easily supports Tailwind whereas Tamagui is more uncertain.

Trying to shed some light on this uncertainty.

—-

[1]

https://twitter.com/magnemg/status/1602993035360493568

https://www.youtube.com/watch?v=hdi8PmeKlbw

From what I understand:
You wrap a React component but sprinkle it with some unique Qwik props like host:onClick , or optionally client:visible (like in Astro).
Either the components run on the server and return HTML which Qwik interweaves into the current page, or you can make them reactive by using signals for state (which doesn't need to run the entire component anywhere to update the state).

**Comment by airtonix** [comment]
@redbar0n does this also mean this a request for Astro support? 

Are there workarounds for now?

**Reply by redbar0n** [reply]
This discussion is for Qwik support specifically. Please open a separate discussion to discuss potential Astro support.

I know of no workarounds, or anyone who have tried to combine Tamagui with Astro, but I'm no insider. I would also ask in the Discord if I were you.
